contracts/exchange/contracts/src/MixinExchangeCore.sol

```diff
@@ -207,8 +207,9 @@ contract MixinExchangeCore is
         );
 
         // Get amount of takerAsset to fill
-        uint256 remainingTakerAssetAmount = _safeSub(order.takerAssetAmount, orderInfo.orderTakerAssetFilledAmount);
-        uint256 takerAssetFilledAmount = _min256(takerAssetFillAmount, remainingTakerAssetAmount);
+        uint256 remainingTakerAssetAmount = order.takerAssetAmount - orderInfo.orderTakerAssetFilledAmount;
+        uint256 takerAssetFilledAmount = takerAssetFillAmount < remainingTakerAssetAmount ? 
```

Comment:
I actually kind of like keeping `min256`, since it's not affected by these changes to how overflows/underflows are handled. Can we move that to `LibMath` and continue to use it?